src/common/vitals/vital-metric.js

```diff
@@ -0,0 +1,56 @@
+export class VitalMetric {
+  #values = []
+  #roundingMethod = Math.floor
+  #subscribers = new Set()
+  entries = undefined
+
+  constructor (name, roundingMethod) {
+    this.name = name
+    this.attrs = {}
+    if (typeof roundingMethod === 'function') this.#roundingMethod = roundingMethod
+  }
+
+  update ({ value, entries, attrs, addConnectionAttributes = false }) {
+    this.#values.push(this.#roundingMethod(value))
+    this.entries = entries
```

Comment:
The old values are kept, then so should the old entries. The new `entries` may not contain the same subset of records as the existing `entries`, so `this.entries` needs to be appended or merged each update too?

Reply:
f6bb821e7788e7027484256438a5ab086de38cce